src/Appwrite/Platform/Workers/Certificates.php

```diff
@@ -187,6 +190,10 @@ private function execute(Domain $domain, Database $dbForConsole, Mail $queueForM
             // Save all changes we made to certificate document into database
             $this->saveCertificateDocument($domain->get(), $certificate, $success, $dbForConsole, $queueForEvents, $queueForFunctions);
```

Comment:
```suggestion
            $this->saveCertificateDocument($domain->get(), $certificate, $success, $dbForConsole, $queueForEvents, $queueForFunctions);
            throw $e;
```

Why not just this?

Reply:
Its in `finally`, we dont have exception there.
I don't want to re-throw inside catch, as I am not sure if finally would run after that properly. And Im afraid to experiment since this is untested piece of code.

Reply:
Hmm or maybe finally does see exception? 🤔

![CleanShot 2024-01-24 at 13 38 59@2x](https://github.com/appwrite/appwrite/assets/19310830/a2d3aebb-8c41-4377-a0df-d3ab1a4b6d0e)

@christyjacob4  can you confirm you are familiar with this behaviour? First time seeing this, worried to use it.

Reply:
![CleanShot 2024-01-24 at 13 40 05@2x](https://github.com/appwrite/appwrite/assets/19310830/a1901ea9-809d-449b-8294-1317a6a5a0e8)

It still causes warning if no exception is there, so it doesn't solve issue with having to do if(isset))

Reply:
Okay then you can do this 
```php
try {
    // some code
} catch (Exception $e) {
  // Handle Error
   throw $e;
} finally {
   // Finally is always executed 
}
```